### ISSUE TITLE
Customers would like to generate JS interop bindings

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/sdk#33715]
(Stub, we can fill in the details as they become relevant)

The popular ones right now are TypeScript definition files and Closure externs, I believe.

**Comment by zoechi** [comment]
There was some work done a while ago https://github.com/dart-lang/sdk/issues/24874#issuecomment-253935419

**Comment by matanlurey** [comment]
Yes, this is a very stale tool. This request is for an updated tool (that is tested on CI, etc).

**Comment by robbecker-wf** [comment]
Are people still interested in such a tool? Workiva is hitting a good deal of pain around JS interop and are considering writing something ourselves.

**Comment by jacobsimionato** [comment]
I'm curious if there have been any discussion around this recently? I suspect this could be very useful for Google apps seeing as nearly all the JS interop that needs to be done is now actually with Typescript. It seems like Typescript has only increased in usage in the 3 years since the last comment, and I understand there is work being done to revisit Dart's JS interop story.

**Comment by parlough** [comment]
@jodinathan has been exploring this space recently with some of the new interop functionality: https://pub.dev/packages/typings
